tests/query/processors/test_hash_map.py

```diff
@@ -0,0 +1,235 @@
+                nested_condition("tags", ConditionFunctions.EQ, "my_tag", "a"),
+                nested_condition("tags", ConditionFunctions.LIKE, "my_tag2", "b"),
+            ),
+        ),
+        binary_condition(
+            None,
+            BooleanFunctions.OR,
+            nested_condition("tags", ConditionFunctions.EQ, "my_tag", "a"),
+            nested_condition("tags", ConditionFunctions.LIKE, "my_tag2", "b"),
+        ),
+        id="Unsupported and supported conditions",
+    ),
+    pytest.param(
+        build_query(
+            selected_columns=[column("event_id")],
+            condition=binary_condition(
+                None,
+                BooleanFunctions.AND,
+                nested_condition("tags", ConditionFunctions.EQ, "my_tag", "a"),
+                binary_condition(
+                    None,
+                    ConditionFunctions.LIKE,
+                    Column(None, None, "something_else"),
+                    Literal(None, "123123"),
+                ),
+            ),
+        ),
+        binary_condition(
+            None,
+            BooleanFunctions.AND,
+            FunctionCall(
+                None,
+                "has",
+                (
+                    column("_tags_hash_map", True),
+                    FunctionCall(None, "cityHash64", (Literal(None, "my_tag=a"),)),
+                ),
+            ),
+            binary_condition(
+                None,
+                ConditionFunctions.LIKE,
+                Column(None, None, "something_else"),
+                Literal(None, "123123"),
+            ),
+        ),
+        id="Supported multiple conditions",
+    ),
+    pytest.param(
+        build_query(
+            selected_columns=[column("event_id")],
+            condition=binary_condition(
+                None,
+                ConditionFunctions.EQ,
+                FunctionCall(
+                    None,
+                    "ifNull",
+                    (nested_expression("tags", "my_tag"), Literal(None, "")),
+                ),
+                Literal(None, ""),
+            ),
+        ),
+        binary_condition(
+            None,
+            ConditionFunctions.EQ,
+            FunctionCall(
+                None,
+                "ifNull",
+                (nested_expression("tags", "my_tag"), Literal(None, "")),
+            ),
+            Literal(None, ""),
+        ),
+        id="Unsupported ifNull condition.",
+    ),
+    pytest.param(
+        build_query(
+            selected_columns=[column("event_id")],
+            condition=nested_condition("tags", ConditionFunctions.EQ, "my_tag", "a"),
+            having=binary_condition(
+                None,
+                ConditionFunctions.EQ,
+                FunctionCall(None, "arrayjoin", (Column(None, None, "tags.key"),)),
+                Literal(None, "bla"),
+            ),
+        ),
+        nested_condition("tags", ConditionFunctions.EQ, "my_tag", "a"),
```

Comment:
Is it worth testing a tag key/value that requires escaping here?

Reply:
you are right. Will add.